### PR TITLE
Update page_data.cu

### PR DESCRIPTION
**Pull request by goodengineer**
batch size multiple definitions extracted to 1 macro; a few small typo fixes; attempt to shrink memory consumption by definint part of intermediate variables as short; ...proposal...

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: cpp/src/io/parquet/page_data.cu [hunk 1]
```diff
@@ -270,7 +270,7 @@ __device__ void gpuDecodeStream(
     }
     if (s->error) { break; }
 
-    batch_len = min(num_input_values - value_count, 32);
+    batch_len = min(num_input_values - value_count, BATCH_SZ); //defined in parquet_common.h
```
Comment (karthikeyann) [inline cpp/src/io/parquet/page_data.cu:273]: `BATCH_SZ` can't be found in https://github.com/goodengineer/cudf/blob/patch-4/cpp/src/io/parquet/parquet_common.h

File: cpp/src/io/parquet/page_data.cu [hunk 2]
```diff
@@ -1200,11 +1200,11 @@ static __device__ void gpuUpdateValidityOffsetsAndRowIndices(int32_t target_inpu
                                                              int t)
 {
   // max nesting depth of the column
-  int max_depth = s->col.max_level[level_type::REPETITION];
+  short max_depth = s->col.max_level[level_type::REPETITION];
   // how many (input) values we've processed in the page so far
   int input_value_count = s->input_value_count;
   // how many rows we've processed in the page so far
-  int input_row_count = s->input_row_count;
+  short input_row_count = s->input_row_count;
```
Comment (karthikeyann) [inline cpp/src/io/parquet/page_data.cu:1207]: Does this shrink memory consumption?
For resource usage, after compilation 
`cuobjdump  --dump-resource-usage ./CMakeFiles/cudf.dir/src/io/parquet/page_data.cu.o | c++filt`
Compare.

**Review by karthikeyann — CHANGES_REQUESTED** [review]
Compare register usage.

**Comment by harrism** [comment]
We have a major overhaul of all of cuIO underway. Since one of the tasks in that overhaul is to eliminate magic numbers and macros, I think we should close this.

**Pull request closed without merging**